tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+    """
+    Struct class to hold the common variables shared between analyze tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'analyze',
+            '--analysis-models',
+            'a,b,c'
+        ]
+        #yapf: enable
+        config_analyze = ConfigCommandAnalyze()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='analyze', help='', config=config_analyze)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class CLIConfigReportStruct():
+    """
+    Struct class to hold the common variables shared between analyze tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'report',
+            '--report-model-configs',
+            'a, b, c'
+        ]
+        #yapf: enable
+        config_report = ConfigCommandReport()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='report', help='', config=config_report)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class OptionStruct():
+
+    def __init__(self,
+                 type_str,
+                 stage,
+                 long_flag,
+                 short_flag=None,
+                 expected_value=None,
+                 default_value=None,
+                 expected_failing_value=None,
+                 extra_commands=None):
+
+        self.long_flag = long_flag
+        self.short_flag = short_flag
+        self.expected_value = expected_value
+        self.default_value = default_value
+        self.expected_failing_value = expected_failing_value
+        self.type = type_str
+        self.extra_commands = extra_commands
+
+        if stage == "profile":
+            self.cli_subcommand = CLIConfigProfileStruct
+        elif stage == "analyze":
+            self.cli_subcommand = CLIConfigAnalyzeStruct
+        elif stage == "report":
+            self.cli_subcommand = CLIConfigReportStruct
+
+
+@patch('model_analyzer.config.input.config_command_profile.file_path_validator',
+       lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+@patch(
+    'model_analyzer.config.input.config_command_profile.binary_path_validator',
+    lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+@patch('model_analyzer.config.input.config_command_analyze.file_path_validator',
+       lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+@patch('model_analyzer.config.input.config_command_report.file_path_validator',
+       lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+class TestCLIOptions(trc.TestResultCollector):
+    """
+    Tests the methods of the CLI class
+    """
+
+    def test_basic_cli_config_profile_options(self):
+        """
+        Test the minimal set of cli commands necessary to run Model Analyzer profile
+        """
+        cli = CLIConfigProfileStruct()
+        _, config = cli.parse()
```

Comment:
Not familiar with this syntax: what does _, do?

Reply:
_ is a assignment placeholder that is immediately thrown away.  The code here `cli.parse()` returns a tuple. I am immediately unpacking the tuple to get the config arguments but i dont care about the args that are returned as part of the tuple. I cannot unpack the tuple and not assign the args so I use the `_` here to assign them but immediately throw them away, thereby avoiding errors and warnings around unused variables.

Reply:
Thanks!